src/files/url-source.js

```diff
@@ -2,12 +2,17 @@
 
 const Http = require('../http')
 
-module.exports = async function * urlSource (url, options) {
-  const http = new Http()
-  const response = await http.get(url, options)
-
-  yield {
+const urlSource = (url, options) => {
+  return {
     path: decodeURIComponent(new URL(url).pathname.split('/').pop() || ''),
-    content: response.iterator()
+    content: readURLContent(url, options)
   }
 }
+
+const readURLContent = async function * (url, options) {
+  const http = new Http()
+  const response = await http.get(url, options)
+  yield * response.iterator()
+}
```

Comment:
i dont understand this change, why the wrapper ?

Reply:
@hugomrdias so previously we had `ipfs.add` that could imported bunch of files or import a single one. Which later we have split into `ipfs.add` that imports a single file and takes single file like thing, and `ipfs.addAll` which imports multilpe files.

This change just makes return of `urlSource` compatible with `ipfs.add` as opposed to `ipfs.addAll`.

P.S.: `ipfs.add` implementation (but not the typings) will still accept many files, but plan is to make that not a case in the future.

Reply:
To answer your specific question, this does not add a wrapper but rather moves code doing the content fetching into `content` field which is expected to be `AsyncIterable<Uint8Array>`. effectively changing return type of this `urlSource` from `AsyncIterable<{ path: string, content: AsyncIterable<Uint8Array> }>` to `{ path: string, content: AsyncIterable<Uint8Array> }` removing one layer of asynchronicity and making return be a single file as opposed to collection of files that always happen to contain one.